src/main/java/edu/kis/powp/jobs2d/drivers/DriverMonitoringDecorator.java

```diff
@@ -0,0 +1,44 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import java.util.logging.Logger;
+
+public class DriverMonitoringDecorator implements Job2dDriver {
+    protected final Job2dDriver driver;
+    private int headDistance = 0;
+    private int opDistance = 0;
```

Comment:
From the point of SRP it would be better to have separate component responsible for the managmement of those values. Those values should be also made accessible (e.g., via getters). 
DriverMonitoringDecorator should mainly trigger changes to those values.